### PR TITLE
Add comprehensive E2E test suite with Playwright

### PR DESCRIPTION
**Pull request by nickmisasi**
## Summary

Adds comprehensive end-to-end test coverage using Playwright with **106 unique test cases** executed across 2 browsers (chromium and firefox). Includes both mock-based tests for fast feedback and real API tests for integration validation.

### System Console Tests (Mock-based)
Complete coverage of the AI plugin's System Console configuration interface:
- **bot-management-ui.spec.ts**: Bot creation and card-based UI configuration
- **bot-native-tools.spec.ts**: Native tool visibility and configuration (Anthropic/OpenAI)
- **bot-reasoning-config.spec.ts**: Reasoning configuration (OpenAI effort levels, Anthropic thinking budgets)
- **bot-validation.spec.ts**: Bot configuration validation badges and error states
- **debug-panel.spec.ts**: Debug toggles (LLM Trace, Token Usage Logging)
- **functions-panel.spec.ts**: AI Functions configuration (default bot, render links)
- **initial-state-navigation.spec.ts**: System Console navigation and initial state
- **service-management-ui.spec.ts**: Service management interface

### Bot Configuration Tests (Mock-based)
- **bot-identity-changes.spec.ts**: Bot display name updates and persistence
- **reasoning-config.spec.ts**: Reasoning configuration integration tests
- **service-changes.spec.ts**: Service provider and configuration changes

### RHS Core Functionality (Mock-based)
- **basic.spec.ts**: Core RHS interactions, bot switching, regeneration, error handling
- **direct-channel-creation.spec.ts**: DM channel creation with bots
- **new-messages-rhs.spec.ts**: New messages line integration

### Feature-Specific Tests (Mock-based)
- **Action Item Extraction**: Basic extraction, edge cases, error handling, follow-ups, UI verification
- **Channel Summarization**: Basic summarization functionality
- **Semantic Search**: Basic search operations
- **Smart Reactions**: Reaction suggestions and error handling
- **Long Conversation Handling**: Pagination, scrolling, UI responsiveness
- **Multiple Bot Conversations**: Bot switching, chat history

### Advanced Scenarios (Mock-based)
- **network-errors.spec.ts**: HTTP error handling (401, 403, 429, 500, 503), system resilience
- **input-validation.spec.ts**: Edge cases, XSS prevention, special content

### Real API Tests (Require API Keys)
Located in `tests/llmbot-post-component/`:
- **combined-features.spec.ts**: Reasoning + citations together, regeneration
- **citations-annotations.spec.ts**: Citation display, tooltips, links, persistence
- **reasoning-display.spec.ts**: Reasoning expand/collapse, persistence, multiple posts
- **streaming-persistence.spec.ts**: Streaming lifecycle, navigation persistence
- **edge-cases.spec.ts**: Edge cases and error handling with real APIs

Requires `ANTHROPIC_API_KEY` and/or `OPENAI_API_KEY` environment variables.

### Helper Utilities
- **system-console-container.ts**: Dedicated container setup for System Console tests with mock configurations
- **system-console.ts**: Page object helpers for System Console navigation and interaction
- **real-api-container.ts**: Container configuration for real API tests
- **llmbot-post.ts**: Page object for LLMBot post component interactions
- **api-config.ts**: API service configuration utilities
- **bot-config.ts**: Bot configuration utilities

### Key Features
- Testcontainers-based infrastructure for isolated test environments
- Mock OpenAI server for fast, deterministic tests
- Cross-browser testing (chromium and firefox)
- Factory pattern for flexible test configuration
- Smart polling with 5-minute safety timeouts
- Proper handling of viewport scrolling and race conditions
- Support for both Anthropic and OpenAI providers
- Reactive UI validation for real-time configuration changes

## Test Execution

Run all mock-based tests (excluding real API tests):
```bash
cd e2e && npx playwright test --grep-invert "llmbot-post-component"
```

Run only System Console tests:
```bash
cd e2e && npx playwright test tests/system-console
```

Run real API tests (requires API keys):
```bash
cd e2e && npx playwright test tests/llmbot-post-component
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: test-results/.last-run.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (crspeller) [inline test-results/.last-run.json]: Does this need to be ignored?

File: .claude/agents/playwright-test-generator.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+---
+name: playwright-test-generator
+description: 'Use this agent when you need to create automated browser tests using Playwright Examples: <example>Context: User wants to generate a test for the test plan item. <test-suite><!-- Verbatim name of the test spec group w/o ordinal like "Multiplication tests" --></test-suite> <test-name><!-- Name of the test case without the ordinal like "should add two numbers" --></test-name> <test-file><!-- Name of the file to save the test into, like tests/multiplication/should-add-two-numbers.spec.ts --></test-file> <seed-file><!-- Seed file path from test plan --></seed-file> <body><!-- Test case content including steps and expectations --></body></example>'
```
Comment (crspeller) [inline .claude/agents/playwright-test-generator.md:3]: Seems like this could be simplified. Not sure what all the tags are about.

File: .claude/agents/playwright-test-generator.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+tools: Glob, Grep, Read, LS, mcp__playwright-test__browser_click, mcp__playwright-test__browser_drag, mcp__playwright-test__browser_evaluate, mcp__playwright-test__browser_file_upload, mcp__playwright-test__browser_handle_dialog, mcp__playwright-test__browser_hover, mcp__playwright-test__browser_navigate, mcp__playwright-test__browser_press_key, mcp__playwright-test__browser_select_option, mcp__playwright-test__browser_snapshot, mcp__playwright-test__browser_type, mcp__playwright-test__browser_verify_element_visible, mcp__playwright-test__browser_verify_list_visible, mcp__playwright-test__browser_verify_text_visible, mcp__playwright-test__browser_verify_value, mcp__playwright-test__browser_wait_for, mcp__playwright-test__generator_read_log, mcp__playwright-test__generator_setup_page, mcp__playwright-test__generator_write_test
+model: sonnet
+color: blue
```
Comment (crspeller) [inline .claude/agents/playwright-test-generator.md:6]: `color: blue`? Does this work?

**Review by crspeller — CHANGES_REQUESTED** [review]
Good stuff! Just a few little comments.

File: .claude/agents/playwright-test-generator.md (hunk 2 above)
Comment (nickmisasi) [reply .claude/agents/playwright-test-generator.md:3]: This is actually not my agent, it comes this way straight from Playwright. 

It's XML prompting with XML comments. I'm inclined to leave it, since all of the testing with the agents I've done with them like this

File: .claude/agents/playwright-test-generator.md (hunk 3 above)
Comment (nickmisasi) [reply .claude/agents/playwright-test-generator.md:6]: Yep - another thing that came from playwright, not myself

File: test-results/.last-run.json (hunk 1 above)
Comment (nickmisasi) [reply test-results/.last-run.json]: I think this was just a one off agent debugging thing. I removed the file.
